app/code/community/Adyen/Payment/controllers/ApplePayController.php

```
@@ -259,6 +284,13 @@ public function sendPaymentAction()
             $customer = Mage::getSingleton('customer/session')->getCustomer();
             $quote = $quote->assignCustomer($customer);
         }
+        else {
+            // set the customer email address
+            if(!isset($payment->shippingContact->emailAddress) || $payment->shippingContact->emailAddress == "") {
+                Mage::throwException(Mage::helper('adyen')->__('Missing email address in payment'));
```

@rikterbeek @cyattilakiss @maassenbas Do we want to throw an exception if the customer shippingContact->emailAddress is empty? I did a guest checkout fresh magento and this kept being raised, can you guys check out?

---

Hi @ScottPowell14 for quest checkout this should not be an issue as you need to fill in a email address as this is set to required, https://github.com/Adyen/adyen-magento/blob/30aa63123bc6d9ef4990055493aed7312d702edc/app/design/frontend/base/default/template/adyen/apple_pay.phtml#L199-L200 were you prompted to fill in the email address on the payment sheet ?

---

@rikterbeek Yes, I debugged and actually `$payment->shippingContact` is null which is why the exception is being thrown. When is this shippingContact data set? Have you tested guest checkout and not see the exception?

---

Hi, I've just tested Magento's one page checkout guest mode on this & @ScottPowell14 is right an exception is raised. From OPC the payment request only contains total & shipping costs. The reason I originally added it was that when checking out using ApplePay from the cart the customer email (from the payment request) was not getting saved into the quote. I've pushed another fix into this branch [here](https://github.com/Adyen/adyen-magento/pull/921/commits/3d7a7c97ea9ce89c7bfbb283932847dd45615bd5).

---

Hi @enru 

Are you able to resolve the conflicts on the PR? This would ensure the contribution is in your name.

Please let me know,
Scott

---

Resolved. Thanks.